crates/bitcoin-da/src/helpers/builders.rs

```diff
@@ -378,17 +518,377 @@ pub fn create_inscription_transactions(
+
+    let mut commit_chunks: Vec<Transaction> = vec![];
+    let mut reveal_chunks: Vec<Transaction> = vec![];
+
+    for body in body.chunks(MAX_TXBODY_SIZE) {
+        let header = TransactionHeaderLightClient {
+            rollup_name,
+            kind: TransactionKindLightClient::ChunkedPart,
+        };
+        let header_bytes = header.to_bytes();
+
+        // start creating inscription content
+        let mut reveal_script_builder = script::Builder::new()
+            .push_x_only_key(&public_key)
+            .push_opcode(OP_CHECKSIGVERIFY)
+            .push_slice(PushBytesBuf::try_from(header_bytes).expect("Cannot push header"))
+            .push_opcode(OP_FALSE)
+            .push_opcode(OP_IF);
         // push body in chunks of 520 bytes
         for chunk in body.chunks(520) {
             reveal_script_builder = reveal_script_builder.push_slice(
                 PushBytesBuf::try_from(chunk.to_vec()).expect("Cannot push body chunk"),
             );
         }
         // push end if
-        reveal_script_builder = reveal_script_builder.push_opcode(OP_ENDIF);
+        let reveal_script = reveal_script_builder.push_opcode(OP_ENDIF).into_script();
+
+        // create spend info for tapscript
+        let taproot_spend_info = TaprootBuilder::new()
+            .add_leaf(0, reveal_script.clone())
+            .expect("Cannot add reveal script to taptree")
+            .finalize(&secp256k1, public_key)
+            .expect("Cannot finalize taptree");
+
+        // create control block for tapscript
+        let control_block = taproot_spend_info
+            .control_block(&(reveal_script.clone(), LeafVersion::TapScript))
+            .expect("Cannot create control block");
+
+        // create commit tx address
+        let commit_tx_address = Address::p2tr(
+            &secp256k1,
+            public_key,
+            taproot_spend_info.merkle_root(),
+            network,
+        );
+
+        let commit_value = (get_size(
+            &[TxIn {
+                previous_output: OutPoint {
+                    txid: Txid::from_byte_array([0; 32]),
+                    vout: 0,
+                },
+                script_sig: script::Builder::new().into_script(),
+                witness: Witness::new(),
+                sequence: Sequence::ENABLE_RBF_NO_LOCKTIME,
+            }],
+            &[TxOut {
+                script_pubkey: recipient.clone().script_pubkey(),
+                value: Amount::from_sat(reveal_value),
+            }],
+            Some(&reveal_script),
+            Some(&control_block),
+        ) as f64
+            * reveal_fee_rate
+            + reveal_value as f64)
+            .ceil() as u64;
+
+        // build commit tx
+        let (unsigned_commit_tx, leftover_utxos) = build_commit_transaction(
+            prev_utxo.clone(),
+            utxos,
+            commit_tx_address.clone(),
+            recipient.clone(),
+            commit_value,
+            commit_fee_rate,
+        )?;
```

Comment:
for commit of chunks we don't really need to chain utxos.

it'd be a good idea to use different utxos as inputs for chunks

and THEN have a chunk aggregator that uses the prev_utxo for it's input.

Reply:
see the image I'll send you on slack

Reply:
we can open an issue and do it later though :)

Reply:
https://github.com/chainwayxyz/citrea/issues/1022